### PR TITLE
feat: ResultBytes

### PR DESCRIPTION
**Pull request by flowchartsman**
- Add ResultBytes to sqlitex to support one-result BLOB queries
- Export ErrNoResults/ErrMultipleResults so the user can react to these
  conditions programmatically.
(ref: #85)

### CODE REVIEW

File: sqlitex/query.go [hunk 1]
```diff
@@ -9,8 +9,10 @@ import (
 	"zombiezen.com/go/sqlite"
 )
 
-var errNoResults = errors.New("sqlite: statement has no results")
-var errMultipleResults = errors.New("sqlite: statement has multiple result rows")
+var (
+	ErrNoResults       = errors.New("sqlite: statement has no results")
+	ErrMultipleResults = errors.New("sqlite: statement has multiple result rows")
+)
```
Comment (zombiezen) [inline sqlitex/query.go:15]: Let's revert this change.

File: sqlitex/query.go [hunk 2]
```diff
@@ -100,3 +102,16 @@ func ResultFloat(stmt *sqlite.Stmt) (float64, error) {
 	}
 	return res, nil
 }
+
+// ResultBytes reads the first column of the first and only row produced by running stmt into buf, returning the number of bytes read.
+// It returns an error if there is not exactly one result row.
+func ResultBytes(stmt *sqlite.Stmt, buf []byte) (int, error) {
```
Comment (zombiezen) [inline sqlitex/query.go:108]: Mind adding a line in the CHANGELOG for this new function? (See surrounding entries for format and please link to this PR.)

**Review by zombiezen — CHANGES_REQUESTED** [review]
I'm surprised this function didn't already exist, so I would be delighted to merge this in. 😄 Just a few tweaks, and we should be good to go. Thanks for the PR!

**Comment by zombiezen** [comment]
Oh, and mind adding a small test for this function?

File: sqlitex/query.go (hunk 1 above)
Comment (flowchartsman) [reply sqlitex/query.go:15]: Can I get some clarification on this? Generally, I believe that sentinel errors should be exported if they have meaningful names and convey meaningful information.  In this case, they would allow the caller the option to distinguish between "the call I explicitly expected to have one result has none" and " the call I explicitly expected to have one result has more than one". 

If the argument is that you deliberately want this information to be opaque, I can get behind that I suppose. The semantics of the methods that might return these errors are clear enough ("one result, or error"). That would make these errors not so much sentinel errors but helpful templates. I just want to make sure I'm clear on the reasoning behind the decision to hide this distinction.

Comment (zombiezen) [reply sqlitex/query.go:15]: Yup, sorry for being terse. I'm asking to revert from the standpoint that `ResultBytes` is useful on its own (and the title/focus of the PR) and I would like to merge it in. I am less sure about exporting the errors at this time, so would like to figure that outside the scope of this PR.

**Review by zombiezen — APPROVED** [review]
I'm carrying this forward with the changes I requested so we can get this in.
